crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1409,37 +1410,20 @@ impl RpcServerConfig {
             http_local_addr = Some(local_addr);
             http_server = Some(server);
         }
-
-        Ok(WsHttpServer {
+        http_handle = http_server
+            .map(|http_server| http_server.start(modules.http.clone().expect("http server error")));
+        ws_handle = ws_server
+            .map(|ws_server| ws_server.start(modules.ws.clone().expect("ws server error")));
+        Ok(RpcServerHandle {
             http_local_addr,
             ws_local_addr,
-            server: WsHttpServers::DifferentPort { http: http_server, ws: ws_server },
+            http: http_handle,
+            ws: ws_handle,
+            ipc_endpoint: None,
+            ipc: None,
             jwt_secret: self.jwt_secret,
         })
     }
-
-    /// Finalize the configuration of the server(s).
-    ///
-    /// This consumes the builder and returns a server.
-    ///
-    /// Note: The server is not started and does nothing unless polled, See also
-    /// [`RpcServer::start`]
-    pub async fn build(mut self, modules: &TransportRpcModules) -> Result<RpcServer, RpcError> {
-        let mut server = RpcServer::empty();
-        server.ws_http = self.build_ws_http(modules).await?;
-
-        if let Some(builder) = self.ipc_server_config {
-            let metrics = modules.ipc.as_ref().map(RpcRequestMetrics::ipc).unwrap_or_default();
-            let ipc_path =
-                self.ipc_endpoint.unwrap_or_else(|| constants::DEFAULT_IPC_ENDPOINT.into());
-            let ipc = builder
-                .set_rpc_middleware(IpcRpcServiceBuilder::new().layer(metrics))
-                .build(ipc_path);
-            server.ipc = Some(ipc);
-        }
```

Comment:
where did the ipc setup go?

Reply:
in my recent changes i've integrated it into the start method, thank you for pointing this out